app/services/create_dqt_trn_request.rb

```diff
@@ -11,12 +11,25 @@ def call
     request_id = SecureRandom.uuid
     dqt_trn_request = DQTTRNRequest.create!(request_id:, application_form:)
 
-    dqt_trn_request.update_from_dqt_response(
-      DQT::Client::CreateTRNRequest.call(request_id:, application_form:),
-    )
+    begin
+      response =
+        DQT::Client::CreateTRNRequest.call(request_id:, application_form:)
+    rescue StandardError
+      dqt_trn_request.destroy!
```

Comment:
I don't think we can do this as we'll risk creating duplicates on DQT. If the request goes through but appears to fail to us, a timeout or network issue on the way back, then we need to retry with the same request_id as that's what the API uses to detect duplicate requests. I think we should split the API request and the creating the DQTTRequest into separate operations. Create the record with the ID and then pass it into the job and on into here. We wouldn't need to worry about the retry then. We might need to think about another state on it though I suppose, like new, pending, complete or something with new indicated we've never sent it yet. Maybe I'm getting carried away there and pending is fine.

Reply:
Yeah, that’s true, the is a risk there. I figure though in that case the retry would still fail as we’d be attempted to create a TRN request with the same ID, but we can handle that and instead perform the read operation.

I had thought about doing it as you suggest but liked the idea of there only being a record in our database if the request has succeeded, but you’re right there is a case where that doesn’t work. I will add a third “new” state to represent this, I think it’ll be required to properly understand where a request got to if we need to retry.

Reply:
Could this service create the DQTTrnRequest record and enqueue a job passing the record as an argument?

Reply:
I was thinking we would pass the request ID in and the service would find or create, just to keep it all within the service but that would work too.